### PR TITLE
 samples: Bluetooth: Update central multilink and peripheral identity

### PR DESCRIPTION
**Pull request by cvinayak**
Update central multilink sample to use continuous scanning
to speed up connection setup.

Update peripheral identity sample to be configurable for
bsim testing as multiple devices with configurable count
of supported connections.

Enable GATT Client, MTU exchange in the multiple_id bsim
test.

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

File: samples/bluetooth/central_multilink/src/central_multilink.c [hunk 1]
```diff
@@ -20,8 +20,8 @@
 #include <zephyr/bluetooth/gatt.h>
 #include <zephyr/sys/byteorder.h>
 
-#define SCAN_INTERVAL 0x0640 /* 1000 ms */
-#define SCAN_WINDOW   0x0030 /* 30 ms */
+#define SCAN_INTERVAL 0x0010 /* 10 ms */
+#define SCAN_WINDOW   0x0010 /* 10 ms */
```
Comment (rugeGerritsen) [inline samples/bluetooth/central_multilink/src/central_multilink.c:24]: We have a macro `BT_LE_SCAN_ACTIVE_CONTINUOUS` which we could consider using

**Review by jhedberg — DISMISSED** [review]

File: samples/bluetooth/central_multilink/src/central_multilink.c [hunk 2]
```diff
@@ -186,7 +187,7 @@ static void disconnected(struct bt_conn *conn, uint8_t reason)
 
 	bt_conn_unref(conn);
 
-	if ((conn_count == 1U) && is_disconnecting) {
+	if ((conn_count == 1U) && (is_disconnecting || (reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB))) {
```
Comment (Thalley) [inline samples/bluetooth/central_multilink/src/central_multilink.c:190]: Doesn't the `reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB` come in the `connected` callback, and not the disconnected callback?

Comment (cvinayak) [reply samples/bluetooth/central_multilink/src/central_multilink.c:190]: First connection_complete with reason success comes in `connected` callback, then 6 connection interval later `reason == BT_HCI_ERR_CONN_FAIL_TO_ESTAB` come in the `disconnected` callback.

I have added new two variants with 20 connections each.

In the many central devices tests, each central tries connecting at the same time to the single advertising peripheral. In bsim, one of 20 central manages to successful get its `CONNECT_IND` PDU to the peripheral, 19 others get "failed to establish" .... this goes on till all 20 centrals make 20 parallel connection to the single peripheral device.

The update in this line is to make the central device to re-scan and connect when it is the unlucky one amongst all centrals that sent the overlapping `CONNECT_IND`.

**Comment by cvinayak** [comment]
@jhedberg  and @Thalley  sorry to have added more to this PR, as the new changes are related and was curious as to CI results with these added coverages.

I am hoping the updates in this PR will cover regressions in Host around connections and disconnection. Running with increased iterations can bring confidences around buffer management (and hopefully catch buffer leaks, if any, in the future).

Adding GATT transactions and SMP procedures not yet in this PR would be next, i am sure it will uncover some deadlocks around local initiated procedures in combination with handling responses to remote initiated procedures.

**Comment by Thalley** [comment]
> Adding GATT transactions and SMP procedures not yet in this PR would be next, i am sure it will uncover some deadlocks around local initiated procedures in combination with handling responses to remote initiated procedures.

You seem to be treating this sample more and more as a test application. I would suggest considering creating a new bsim test application, or extending on our of other bsim test applications, for such things. Samples are not test applications, and should ideally be simple. 

That being said, I'm not strictly opposed to enabling security and some GATT operations in these samples, as that can be useful when showcasing that we can setup and use multiple multiple links, which I believe is the purpose of the sample

File: samples/bluetooth/peripheral_identity/src/peripheral_identity.c [hunk 3]
```diff
@@ -264,25 +283,33 @@ int init_peripheral(uint8_t iterations)
...
 	uint8_t prev_count = conn_count;
 	while (1) {
 		/* If maximum connections is reached, wait for disconnections
-		 * initiated by peer central devices.
 		 */
-		if (conn_count == CONFIG_BT_MAX_CONN) {
+		if (conn_count == conn_count_max) {
+			is_disconnecting = true;
+
+			k_sleep(K_SECONDS(60));
```
Comment (Thalley) [inline samples/bluetooth/peripheral_identity/src/peripheral_identity.c:298]: Why do have a 60 second sleep here?

File: tests/bsim/bluetooth/ll/multiple_id/tests_scripts/multiple_central.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+simulation_id="central_single_peripheral_multilink"
+verbosity_level=2
+EXECUTE_TIMEOUT=1600
+
+cd ${BSIM_OUT_PATH}/bin
+
+Execute ./bs_${BOARD_TS}_tests_bsim_bluetooth_ll_multiple_id_prj_conf \
+  -v=${verbosity_level} -s=${simulation_id} -d=0 -testid=peripheral_multilink
+
+peripheral_count=20
```
Comment (Thalley) [inline tests/bsim/bluetooth/ll/multiple_id/tests_scripts/multiple_central.sh:17]: Suggest to use `peripheral_count` when setting `-D=21` - That way the value can be modified in a single place

Comment (cvinayak) [reply tests/bsim/bluetooth/ll/multiple_id/tests_scripts/multiple_central.sh]: done

**Comment by cvinayak** [comment]
> You seem to be treating this sample more and more as a test application.

I am designing the sample for test here; GATT and SMP are core Bluetooth features, having them does not make it a test application, but a correct sample that discovers services and is secure.

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**